src/os/bluestore/StupidAllocator.cc

```diff
@@ -193,6 +193,8 @@ int64_t StupidAllocator::allocate(
   if (max_alloc_size == 0) {
     max_alloc_size = want_size;
   }
+  // cap with 32-bit val
+  max_alloc_size = std::min(max_alloc_size, 0x10000000 - alloc_unit);
```

Comment:
i don't understand why we need to be alloc_unit smaller than 2^32... isn't that what we end up with anyway if we're < 2^32 and any allocation is already a multiple of alloc_unit?

Reply:
well, this part is actually optional - I faced an infinite loop while writing test case for the previous commit - an attempt to allocate 4GB stuck for ever due to 32-bit length returned from allocate_int.
Hence decided to fix this as well...